parallelcluster/add-external-lb.sh

```diff
@@ -1,18 +1,43 @@
 #!/bin/bash
-export cluster="security3"
-tempfile=`mktemp`
-aws elbv2 describe-tags --resource-arns $(aws elbv2 describe-target-groups --query 'TargetGroups[*].TargetGroupArn' --output text) --query "TagDescriptions[?Tags[?Key==\`parallelcluster:cluster-name\` && Value==\`$cluster\`]].ResourceArn" --output text > $tempfile
-target_group_arn=`cat $tempfile`
+export cluster="security"
+
+# Find existing internal LB ARN
+tempfile=$(mktemp)
+aws elbv2 describe-tags --resource-arns $(aws elbv2 describe-load-balancers --query 'LoadBalancers[*].LoadBalancerArn' --output text) \
+--query "TagDescriptions[?Tags[?Key=='parallelcluster:cluster-name' && Value=='$cluster']].ResourceArn" --output text > $tempfile
+internal_lb_arn=$(cat $tempfile)
+rm $tempfile
+
+# DNS Name of internal LB 
+internal_lb_dns=`aws elbv2 describe-load-balancers --load-balancer-arns $internal_lb_arn --query 'LoadBalancers[0].DNSName' --output text`
+
+# We need the head node id to inquire the IP address of the internal LB (the AWS API does not seem to expose the IP of a NLB)
+head_node_id=`pcluster describe-cluster --cluster-name="$cluster" | jq -r .headNode.instanceId`
+
+command="host $internal_lb_dns | awk '{print \$4}'"
+
+command_id=`aws ssm send-command --instance-ids "$head_node_id" \
+    --document-name "AWS-RunShellScript" \
+    --parameters 'commands=["'"$command"'"]' | jq -r .Command.CommandId`
+
+# IP address of internal NLB
+internal_lb_ip=`aws ssm get-command-invocation \
+    --command-id "$command_id" \
+    --instance-id "$head_node_id" | jq -r .StandardOutputContent`
 
 # Query VPC ID
-vpc_id=`aws elbv2 describe-target-groups --target-group-arns $target_group_arn  --query 'TargetGroups[0].VpcId' --output text`
+vpc_id=`aws elbv2 describe-load-balancers --load-balancer-arns $internal_lb_arn --query 'LoadBalancers[0].VpcId' --output text`
 
-# Add new SG
-aws ec2 create-security-group --group-name my-security-group-$cluster --description "Security group for port 8787 access from everywhere" --vpc-id $vpc_id
+# Create new Target group 
```

Comment:
Can we create this load balancer, target group, and so ahead of time in Pulumi? Then just do the registration here? That way, the majority of the infrastructure can come down with pulumi, and we don't accidentally leave cluster remnants behind. I'm happy to work on this as a separate PR if that works for you.

Reply:
Can do!

Reply:
Please keep me in the loop once you get started so we avoid any double programming.